services/scan/src/precinct_scanner_app_other.test.ts

```diff
@@ -0,0 +1,393 @@
+const ballotImages = {
+  completeHmpb: [
+    electionFamousNames2021Fixtures.handMarkedBallotCompletePage1.asFilePath(),
+    electionFamousNames2021Fixtures.handMarkedBallotCompletePage2.asFilePath(),
+  ],
+  completeBmd: [
+    electionFamousNames2021Fixtures.machineMarkedBallotPage1.asFilePath(),
+    electionFamousNames2021Fixtures.machineMarkedBallotPage2.asFilePath(),
+  ],
+  unmarkedHmpb: [
+    electionFamousNames2021Fixtures.handMarkedBallotUnmarkedPage1.asFilePath(),
+    electionFamousNames2021Fixtures.handMarkedBallotUnmarkedPage2.asFilePath(),
+  ],
+  wrongElection: [
+    // A BMD ballot front from a different election
+    join(sampleBallotImagesPath, 'sample-batch-1-ballot-1.png'),
+    // Blank BMD ballot back
+    electionFamousNames2021Fixtures.machineMarkedBallotPage2.asFilePath(),
+  ],
+  // The interpreter expects two different image files, so we use two
+  // different blank page images
+  blankSheet: [
+    join(sampleBallotImagesPath, 'blank-page.png'),
+    // Blank BMD ballot back
+    electionFamousNames2021Fixtures.machineMarkedBallotPage2.asFilePath(),
+  ],
+} as const;
+
+async function configureApp(
+  app: Application,
+  {
+    addTemplates = false,
+    precinctId,
+  }: { addTemplates?: boolean; precinctId?: PrecinctId } = {
+    addTemplates: false,
+  }
+) {
+  const { ballots, electionDefinition } = await readBallotPackageFromBuffer(
+    electionFamousNames2021Fixtures.ballotPackage.asBuffer()
+  );
+  await patch(
+    app,
+    '/precinct-scanner/config/election',
+    electionDefinition.electionData
+  );
+  if (addTemplates) {
+    // It takes about a second per template, so we only do some
+    for (const ballot of ballots.slice(0, 2)) {
+      await postTemplate(app, '/precinct-scanner/config/addTemplates', ballot);
+    }
+  }
+  await setAppPrecinct(app, precinctId);
+  await put(app, '/precinct-scanner/electionSession', { testMode: false });
+  await setPollsState(app, 'polls_open');
+}
+
+async function scanBallot(
+  mockPlustek: MockScannerClient,
+  app: Application,
+  initialBallotsCounted: number
+) {
+  await mockPlustek.simulateLoadSheet(ballotImages.completeBmd);
+  await waitForStatus(app, {
+    state: 'ready_to_scan',
+    ballotsCounted: initialBallotsCounted,
+  });
+
+  const interpretation: Scan.SheetInterpretation = {
+    type: 'ValidSheet',
+  };
+
+  await post(app, '/precinct-scanner/scanner/scan');
+  await waitForStatus(app, {
+    state: 'ready_to_accept',
+    interpretation,
+    ballotsCounted: initialBallotsCounted,
+  });
+  await post(app, '/precinct-scanner/scanner/accept');
+  await waitForStatus(app, {
+    ballotsCounted: initialBallotsCounted + 1,
+    state: 'accepted',
+    interpretation,
+  });
+
+  // Wait for transition back to no paper
+  await waitForStatus(app, {
+    state: 'no_paper',
+    ballotsCounted: initialBallotsCounted + 1,
+  });
+}
+
+test("setting the election also sets precinct if there's only one", async () => {
```

Comment:
I wasn't sure what to name this file, but it seemed like `precinct_scanner_app.test` was getting too big, and there was a distinction between the testing of all the scan behavior and then the testing of all the other backend functions. If you agree we can start splitting this testing, I'll extract the helpers above.